src/ethereum/spurious_dragon/vm/interpreter.py

```diff
@@ -0,0 +1,357 @@
+        gas_left=message.gas,
+        env=env,
+        valid_jump_destinations=valid_jump_destinations,
+        logs=(),
+        refund_counter=U256(0),
+        running=True,
+        message=message,
+        output=b"",
+        accounts_to_delete=dict(),
+        has_erred=False,
+        children=[],
+    )
+    try:
+
+        if evm.message.code_address in PRE_COMPILED_CONTRACTS:
+            PRE_COMPILED_CONTRACTS[evm.message.code_address](evm)
+            return evm
+
+        while evm.running and evm.pc < len(evm.code):
+            try:
+                op = Ops(evm.code[evm.pc])
+            except ValueError:
+                raise InvalidOpcode(evm.code[evm.pc])
+
+            op_implementation[op](evm)
+
+    except (
+        OutOfGasError,
+        InvalidOpcode,
+        InvalidJumpDestError,
+        InsufficientFunds,
+        StackOverflowError,
+        StackUnderflowError,
+        StackDepthLimitError,
+    ):
+        evm.gas_left = U256(0)
+        evm.logs = ()
+        evm.accounts_to_delete = dict()
+        evm.refund_counter = U256(0)
+        evm.has_erred = True
+    except (
+        EnsureError,
+        ValueError,
+    ):
+        evm.has_erred = True
+    finally:
+        return evm
+
+
+def collect_touched_accounts(
+    evm: Evm, ancestor_had_error: bool = False
+) -> Iterable[Address]:
+    """
+    Collect all of the accounts that *may* need to be deleted based on
+    `EIP-161 <https://eips.ethereum.org/EIPS/eip-161>`_.
+    Checking whether they *do* need to be deleted happens in the caller.
+    See also: https://github.com/ethereum/EIPs/issues/716
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+    ancestor_had_error :
+        True if the ancestors of the evm object erred else False
+
+    Returns
+    -------
+    touched_accounts: `typing.Iterable`
+        returns all the accounts that were touched and may need to be deleted.
+    """
+    # collect the coinbase account if it was touched via zero-fee transfer
+    if (evm.message.caller == evm.env.origin) and evm.env.gas_price == 0:
+        yield evm.env.coinbase
+
+    # collect those explicitly marked for deletion
+    # ("beneficiary" is of SELFDESTRUCT)
+    for beneficiary in sorted(set(evm.accounts_to_delete.values())):
+        if evm.has_erred or ancestor_had_error:
+            # Special case to account for geth+parity bug
+            # https://github.com/ethereum/EIPs/issues/716
+            if beneficiary == THREE:
+                yield beneficiary
+            continue
+        else:
+            yield beneficiary
+
+    # collect account directly addressed
+    if evm.message.target != Bytes0(b""):
+        if evm.has_erred or ancestor_had_error:
+            # collect RIPEMD160 precompile even if ancestor evm had error.
+            # otherwise, skip collection from children of erred-out evm objects
+            if evm.message.target == THREE:
+                # mypy is a little dumb;
```

Comment:
Maybe we can get rid of this mypy error if change line 307 with:
`if evm.message.target != Bytes0(b"")` --> `isinstance(evm.message.target, Bytes0)`
(Haven't tested though)

Reply:
My bad, You're right.